packages/@uppy/core/src/index.js

```diff
@@ -76,13 +76,19 @@ class Uppy {
       meta: {},
       onBeforeFileAdded: (currentFile, files) => currentFile,
       onBeforeUpload: (files) => files,
-      store: DefaultStore()
+      store: DefaultStore(),
+      logger: nullLogger
     }
 
     // Merge default options with the ones set by user
     this.opts = Object.assign({}, defaultOptions, opts)
     this.opts.restrictions = Object.assign({}, defaultOptions.restrictions, this.opts.restrictions)
 
+    // Support debug: true for backwards-compatability
+    if (this.opts.debug) {
```

Comment:
If you already set a custom logger, but also debug:true should we refrain? throw a warning/error?

Reply:
(spoiler, i think yes, because coursera might want to ship their logs to a remote place, and then someone else thinks  hey now i want debug level of those logs and then nothing is exported to that server anymore, in a pretty silent way)

Reply:
Good point! Updated, though logging a warning in this case is bit weird, because logging goes to your custom logger, which might be silent in console :)